modules/virt-generalizing-linux-vm-image.adoc

```diff
@@ -0,0 +1,97 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/creating_vms_custom/virt-creating-vms-uploading-images.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-generalizing-linux-vm-image_{context}"]
+= Generalizing a VM image
+
+You can generalize a {op-system-base-full} image to remove all system-specific configuration data before you use the image to create a golden image, a preconfigured snapshot of a virtual machine (VM). You can use a golden image to deploy new VMs.
+
+You can generalize a {op-system-base} VM by using using the `virtctl`, `guestfs`, and `virt-sysprep` tools.
+
+.Prerequisites
+
+* You have an {op-system-base} virtual machine (VM) to use as a base VM.
+* You have installed the OpenShift CLI (`oc`).
+* You have installed the `virtctl` tool.
+
+.Procedure
+
+. Stop the {op-system-base} VM if it is running, by entering the following command:
++
+[source,terminal]
+----
+$ virtctl stop <rhel_vm_name>
+----
+
+. Optional: Clone the virtual machine to avoid losing the data from your original VM. You can then generalize the cloned VM.
+
+. Retrieve the `dataVolume` that stores the root filesystem for the VM by running the following command:
++
+[source,terminal]
+----
+$ oc get vm <my_vm_name> -o jsonpath="{.spec.template.spec.volumes}{'\n'}"
+
+[{"dataVolume":{"name":"<my_vm_volume>"},"name":"rootdisk"},{"cloudInitNoCloud":{...}]
+----
+
+. Retrieve the persistent volume claim (PVC) that matches the listed `dataVolume` by running the followimg command:
++
+[source,terminal]
+----
+$ oc get pvc
+----
++
+.Example output
+[source,terminal]
+----
+NAME            STATUS   VOLUME  CAPACITY   ACCESS MODES  STORAGECLASS     AGE
+<my_vm_volume> Bound  …
+----
+
+[NOTE]
+====
+If your cluster configuration does not enable you to clone a VM, to avoid losing the data from your original VM, you can clone the VM PVC to a data volume instead. You can then use the cloned PVC to create a golden image.
+
+If you are creating a golden image by cloning a PVC, continue with the next steps, using the cloned PVC.
+====
+
+[start=5]
+. Deploy an interactive container with the PVC and `libguestfs-tools` attached to it by running the following command:
++
+[source,terminal]
+----
+$ virtctl guestfs <my-vm-volume> --uid 107
```

Comment:
this is the my_vm_volume which was shown in the output from ` oc get pvc`

Reply:
Is that also the my_vm_volume from 
`oc get vm <my_vm_name> -o jsonpath="{.spec.template.spec.volumes}{'\n'}"`
?

Reply:
@jcanocan can you help me here answering this question?

Reply:
The command: 

`oc get vm <my_vm_name> -o jsonpath="{.spec.template.spec.volumes}{'\n'}"`

gives you the `DataVolume` used by the VM to storage the filesystem, and the command:

`oc get pvc`

gives you the underlying PVC used by that `DataVolume`.

Normally, both outputs should be the same. The reason to ask the user to run both commands is for safety reasons (it might include an unexpected "_", "-", etc. symbol). The user should use the matching output from `oc get pvcs` command.

Sorry about the confusion. My bad :(